src/Tests/Microsoft.Diagnostics.Monitoring.ConfigurationSchema/SchemaGenerator.cs

```diff
@@ -29,11 +38,62 @@ public string GenerateSchema()
+            JsonSchema systemdConsoleLoggerOptionsSchema = GenerateConsoleLoggerOptionsSchema(systemdConsoleFormatterOptions, ConsoleLoggerFormat.Systemd);
+            JsonSchema defaultConsoleLoggerOptionsSchema = GenerateDefaultConsoleLoggerOptionsSchema(simpleConsoleFormatterOptions);
+
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(jsonConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(simpleConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(systemdConsoleLoggerOptionsSchema);
+            schema.Definitions[nameof(ConsoleLoggerOptions)].AnyOf.Add(defaultConsoleLoggerOptionsSchema);
+
             string schemaPayload = schema.ToJson();
 
             //Normalize newlines embedded into json
             schemaPayload = schemaPayload.Replace(@"\r\n", @"\n", StringComparison.Ordinal);
             return schemaPayload;
         }
+
+        public static JsonSchema GenerateConsoleLoggerOptionsSchema(JsonSchema consoleFormatterOptions, ConsoleLoggerFormat consoleLoggerFormat)
+        {
+            JsonSchema consoleLoggerOptionsSchema = new JsonSchema();
+
+            JsonSchemaProperty formatterNameProperty = new JsonSchemaProperty();
+            JsonSchemaProperty formatterOptionsProperty = new JsonSchemaProperty();
+
+            JsonSchema formatterOptionsSchema = new JsonSchema();
+            formatterOptionsSchema.Reference = consoleFormatterOptions;
+
+            formatterOptionsProperty.OneOf.Add(formatterOptionsSchema);
+
+            formatterNameProperty.ExtensionData = new Dictionary<string, object>();
+            formatterNameProperty.ExtensionData.Add("const", consoleLoggerFormat.ToString());
+
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterName), formatterNameProperty);
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.FormatterOptions), formatterOptionsProperty);
+            consoleLoggerOptionsSchema.RequiredProperties.Add(nameof(ConsoleLoggerOptions.FormatterName));
+
+            return consoleLoggerOptionsSchema;
+        }
+
+        public static JsonSchema GenerateDefaultConsoleLoggerOptionsSchema(JsonSchema consoleFormatterOptions)
+        {
+            JsonSchema consoleLoggerOptionsSchema = new JsonSchema();
+
+            JsonSchemaProperty formatterOptionsProperty = new JsonSchemaProperty();
+            JsonSchemaProperty logToStandardErrorThresholdPropertyDefault = new JsonSchemaProperty();
+
+            JsonSchema formatterOptionsSchema = new JsonSchema();
+            formatterOptionsSchema.Reference = consoleFormatterOptions;
+
+            formatterOptionsProperty.OneOf.Add(formatterOptionsSchema);
+
+            consoleLoggerOptionsSchema.Properties.Add(nameof(ConsoleLoggerOptions.LogToStandardErrorThreshold), logToStandardErrorThresholdPropertyDefault);
```

Comment:
This shouldn't be necessary, correct?

Reply:
I added this (and the other line about allowing additional properties) to essentially require that FormatterName not be provided. I wasn't sure how to require FormatterName not be present, so I just explicitly said which properties could be there and disallowed other properties. Feel free to let me know if you have a more elegant solution in mind :)

Reply:
This seems to work:

```
"ConsoleLoggerOptions": {
    "type": "object",
    "additionalProperties": false,
    "properties": {
        "FormatterName": {
            "description": "Name of the log message formatter to use.",
            "oneOf": [
                {
                    "type": "null"
                },
                {
                    "$ref": "#/definitions/ConsoleLoggerFormat"
                }
            ]
        },
        "LogToStandardErrorThreshold": {
            "description": "The minimum level of messages that are written to Console.Error.",
            "oneOf": [
                {
                    "$ref": "#/definitions/LogLevel"
                }
            ]
        },
        "FormatterOptions": {
            "description": "Formatter options for the built-in console log formatter.",
            "oneOf": [
                {},
                {
                    "type": "null"
                }
            ]
        }
    },
    "oneOf": [
        {
            "required": [
                "FormatterName"
            ],
            "properties": {
                "FormatterName": {
                    "const": "Json"
                },
                "FormatterOptions": {
                    "oneOf": [
                        {
                            "$ref": "#/definitions/JsonConsoleFormatterOptions"
                        }
                    ]
                }
            }
        },
        {
            "required": [
                "FormatterName"
            ],
            "properties": {
                "FormatterName": {
                    "const": "Simple"
                },
                "FormatterOptions": {
                    "oneOf": [
                        {
                            "$ref": "#/definitions/SimpleConsoleFormatterOptions"
                        }
                    ]
                }
            }
        },
        {
            "required": [
                "FormatterName"
            ],
            "properties": {
                "FormatterName": {
                    "const": "Systemd"
                },
                "FormatterOptions": {
                    "oneOf": [
                        {
                            "$ref": "#/definitions/ConsoleFormatterOptions"
                        }
                    ]
                }
            }
        },
        {
            "properties": {
                "FormatterName": {
                    "type": "null"
                },
                "FormatterOptions": {
                    "oneOf": [
                        {
                            "$ref": "#/definitions/SimpleConsoleFormatterOptions"
                        }
                    ]
                }
            }
        }
    ]
}
```